geniza/corpus/views.py

```diff
@@ -304,6 +305,27 @@ def get_queryset(self, *args, **kwargs):
     def get_context_data(self, **kwargs):
         """extend context data to add page metadata"""
         context_data = super().get_context_data(**kwargs)
+        document = self.get_object()
+
+        # get unique set of canvases on this document from images and transcriptions
+        images = (
+            document.iiif_images()
+        )  # order override should be respected by iiif_images()
+        for ed in document.digital_editions().all():
+            for canvas_uri in ed.content_html.keys():
+                if canvas_uri not in [i["canvas"] for i in images]:
+                    # append placeholder image for each canvas not in iiif_images
+                    images.append(
+                        {
+                            "image": {
+                                "info": static("img/ui/all/all/image-unavailable.png"),
+                            },
+                            "label": "",
+                            "shelfmark": "",
+                            "canvas": canvas_uri,
+                            "excluded": False,
+                        }
+                    )
```

Comment:
This code seems more complicated than necessary. I think it's because of the way we're structuring our iiif image return, and I failed to communicate that I wanted to make this a `dict` instead of a list.

Would it make sense to define the placeholder image once outside the view? At least the image path should be defined somewhere more obvious, not buried in the method. (It doesn't work with the current canvas uri structure, let me comment on that separately). Empty values aren't necessary, and probably not even the `False`; django templates will ignore and not error on missing attributes.